### DISCUSSION TITLE
execute forceRef request before pre-execution code?

### DISCUSSION THREAD

**Discussion by nrail-joerg-walter** [AnWeber/httpyac#525] [General]
Is there a way to run a request #2 referenced by forceRef in #1 before the pre-execution of #1?

It seems to me not to be the case. This would be quite useful

**Comment by AnWeber** [comment]
Can you please provide an example what is not working as intended? What do you mean with pre-execution? 

In this example request2 is executed before request1, if I send request1
```
# @name request2
GET https://httpbin.org/anything?id={{$guid}}

###
# @name request1
# @forceRef request2
POST https://httpbin.org/anything
{{json.args.id}}
```

**Comment by nrail-joerg-walter** [comment]
```
# @name request2
GET https://httpbin.org/anything?id={{$guid}}

###
{{
  console.log(!request2);
}}
# @name request1
# @forceRef request2
{{
  console.log(request2);
}}
POST https://httpbin.org/anything
{{json.args.id}}
```
Aaaah. I can do this :-)

Sorry. non-issue. And Thanks for this great software!

**Comment by AnWeber** [comment]
Of course. I'm just building a processing queue there. But some commands do not necessarily respect this, like `@name` or using `?? status`.
